extends Control

@onready var body: TextureRect = %Body
@onready var expression: TextureRect = %Expression
@onready var row_bodies: HBoxContainer = $VBoxContainer/RowBodies
@onready var row_expressions: HBoxContainer = $VBoxContainer/RowExpressions

var bodies := {
	"sophia": preload("res://assets/sophia.png"),
	"pink": preload("res://assets/pink.png")
}

var expressions := {
	"happy": preload("res://assets/emotion_happy.png"),
	"regular": preload("res://assets/emotion_regular.png"),
	"sad": preload("res://assets/emotion_sad.png")
}

func create_button_pink() -> void: 
	var button := Button.new()
	row_bodies.add_child(button)
	var key := "pink"
	button.text = key.capitalize()
	button.pressed.connect(func() -> void:
		body.texture = bodies[key]
	)

func _ready() -> void:
	create_button_pink()
